R/content-tools.R

```diff
@@ -44,35 +35,58 @@ extract_tool_requests <- function(contents) {
   contents[is_tool_request]
 }
 
+as_content_tool_result_factory <- function(request) {
+  force(request)
+
+  function(result = NULL, error = NULL) {
+    if (S7_inherits(result, ContentToolResult)) {
+      result@id <- request@id
+      result@request <- request
+      return(result)
+    }
+
+    ContentToolResult(
+      value = result,
+      error = error,
+      id = request@id,
+      request = request
```

Comment:
Technically, `ContentToolResult` doesn't need `@id` anymore, we could get it from `@request@id`. It seems like it could be helpful to have though, so I kept it for now. Happy to refactor it out if you prefer.

Reply:
I don't feel super strongly about this, but I would generally steer away from having one piece of information stored in two places, as it has the potential to create a head scratching bug in the future.